smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/assembly/CleanupContextPropagation.java

```diff
@@ -0,0 +1,28 @@
+package io.smallrye.reactive.messaging.assembly;
+
+import java.util.List;
+import java.util.concurrent.Executor;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.eclipse.microprofile.context.ThreadContext;
+
+import io.smallrye.reactive.messaging.ChannelRegistar;
+import io.smallrye.reactive.messaging.ChannelRegistry;
+
+@ApplicationScoped
+public class CleanupContextPropagation implements AssemblyHook {
+    @Override
+    public Executor before(List<ChannelRegistar> registars,
+            ChannelRegistry registry) {
+
+        try {
+            CleanupContextPropagation.class.getClassLoader().loadClass("org.eclipse.microprofile.context.ThreadContext");
+            System.out.println("Context Propgation hook called and clearing context...");
+            return ThreadContext.builder().unchanged().propagated().cleared(ThreadContext.ALL_REMAINING).build().currentContextExecutor();
```

Comment:
```suggestion
            return ThreadContext.builder()
                .unchanged()
                .propagated(ThreadContext.APPLICATION)
                .cleared(ThreadContext.ALL_REMAINING)
                .build()
                .currentContextExecutor();
```
It seems we need to propagate the "Application" context. This fixes the error in https://github.com/smallrye/smallrye-reactive-messaging/pull/716/files#r481291280.

What was happening is that context propagation ends up calling ThreadContext.clearedContext() for all the cleared contexts. For Application this ends up here: https://github.com/kabir/jboss-eap7/blob/xp2-reactive/microprofile/context-propagation/extension/src/main/java/org/wildfly/extension/microprofile/context/propagation/providers/ApplicationContextProvider.java#L66 which sets the classloader to the system classloader. (I believe I stole this from OpenLiberty).

With this it looks like my tests now pass :)

But at the same time, I am not sure if this is really, totally 'turning off' context propagation. By that I mean that we will still end up having the full list of ThreadContext providers called on every invocation. But still that might be good enough, I don't know. I think the "Application" (i.e. TCCL) context should be propagated. What might be ideal would be some way to disable a set of providers, so that they don't get called at all.

Reply:
> What might be ideal would be some way to disable a set of providers, so that they don't get called at all.

Actually, perhaps that would be done by calling
```
.unchanged(ALL_REMAINING).propagated().cleared()
```
I need to go now though, will look in the morning